File: src/lessons/welcome/MethodsArgsEntity.java
package lessons.welcome;

import jlm.universe.Direction;

public class MethodsArgsEntity extends universe.bugglequest.SimpleBuggle {
	@Override
	public void forward(int i)  { 
		throw new RuntimeException("Pas le droit d'utiliser forward(int) dans cet exercice");
	}

	@Override
	public void backward(int i) {
		throw new RuntimeException("Pas le droit d'utiliser backward(int) dans cet exercice");
	}


	@Override
	public void run() { 
		move(getY(),getDirection().equals(Direction.NORTH)); 
	} 

	/* BEGIN SOLUTION */
	public void move(int nbPas, boolean forward) {
		if (forward) {
			for (int i=0; i<nbPas; i++) 
				forward();
		} else {
			for (int i=0; i<nbPas; i++) 
				backward();
		}
	}
	/* END TEMPLATE */
}

File: resources/exercises/methods/args/MethodsArgsEntity.py
def forward(i=-1):
    if i!=-1:
        errorMsg("forward(int) forbidden in this exercise")
    else:
        entity.forward()

def backward(i=-1):
    if i!=-1:
        errorMsg("backward(int) forbidden in this exercise")
    else:
        entity.backward()

# BEGIN SOLUTION 
def move(nbPas, goForward):
    if goForward:
        for i in range(nbPas):
            forward()
    else:
        for i in range(nbPas):
            backward()
# END SOLUTION

move(getY(), entity.getDirection() == Direction.NORTH) 
